presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimate.java

```diff
@@ -123,6 +128,11 @@ public SymbolStatsEstimate getSymbolStatistics(Symbol symbol)
         return symbolStatistics.keySet();
     }
 
+    public boolean isUnknown()
+    {
+        return isNaN(outputRowCount);
```

Comment:
Rename the method -- even if we know a lot about stats (eg NDVs for all symbols), this will still say "is unknown", which is misleading.

Reply:
Yes, you are right. I will rename it to `isOutputRowCountUnknown()`

Reply:
@arhimondr I don't think that global `isUnknown` method is a good idea. This is because unknown means different things depending on context (e.g: we might be interested in single column or at plan node stats as a whole).

This method brings yet another concept of unknown stats. We now have:
* `PlanNodeStatsEstimate#isUnknown()` (`PlanNodeStatsEstimate` scope)
* `Optional<PlanNodeStatsEstimate>` in `FilterStatsCalculator` (`PlanNodeStatsEstimate` scope)
* plain `NaN`s (column scope, row count)

What if we just use `NaN`s and each place where we want to widen unknown scope (e.g: from `unknown column` to  `unknown PlanNodeStatsEstimate` or from `unknown row count` to `unknown PlanNodeStatsEstimate` has a dedicated check). For example `FilterExpressionStatsCalculatingVisitor#visitLogicalBinaryAnd` could check explicitly if any composite stats is unknown?

In any case. Changes here seem to exceed scope of this PR `Avoid making wild guesses in FILTER and JOIN estimates`.

Reply:
@sopel39 

>  I don't think that global isUnknown method is a good idea.

I renamed this method to `isOutputRowCountUnknown`. In most cases if the row count is unknown, we don't care if we know anything about single symbol statistics.

> Optional<PlanNodeStatsEstimate> in FilterStatsCalculator

I'm removing this

> Or PlanPrinter could explicitly check if output row count is NaN

I think it is better to have `isOutputRowCountUnknown` then checking for `NaN` everywhere. It is better to encapsulate this special case inside of the `PlanNodeStatsEstimate`.


> In any case. Changes here seem to exceed scope of this PR Avoid making wild guesses in FILTER and JOIN estimates.

I extrated them into separate commits. We can merge them commit by commit.